src/bundle/bundle.ts

```diff
@@ -24,54 +33,78 @@ export interface IBundleProps {
   bundleConfig?: IOutputBundleConfigAdvanced;
   ctx: Context;
   entries?: Array<IModule>;
+  fileName?: string;
   includeAPI?: boolean;
-  type?: IBundleType;
+  priority?: number;
+  type?: BundleType;
+  webIndexed?: boolean;
 }
 
 export interface IBundleWriteResponse {
   absPath: string;
-  bundle?: IBundle;
+  bundle?: Bundle;
   relativePath: string;
 }
 
-export function Bundle(props: IBundleProps): IBundle {
-  const { bundleConfig, ctx } = props;
+export function createBundle(props: IBundleProps): Bundle {
+  const { bundleConfig, ctx, priority, type, webIndexed = true } = props;
   const outputConfig = ctx.outputConfig;
   const isProduction = ctx.config.isProduction;
   const target = ctx.config.target;
 
-  const bundleWriter = distWriter({ hashEnabled: isProduction, root: outputConfig.root });
+  const bundleWriter = distWriter({ hashEnabled: isProduction, root: outputConfig.distRoot });
   const apiCore = props.includeAPI && {
     interopRequireDefault: ctx.compilerOptions.esModuleInterop,
     isIsolated: bundleConfig.isolatedApi,
     target,
   };
-  const source = bundleSource({ core: apiCore, isIsolated: bundleConfig.isolatedApi, target });
+  const source = createBundleSource({ core: apiCore, isIsolated: bundleConfig.isolatedApi, target });
```

Comment:
I would move it inside the `generate` function which accepts `apiCore` paramater